### PR TITLE
Channel switching - https://bugzilla.mozilla.org/show_bug.cgi?id=1132079

### PR DESCRIPTION
**Pull request by trink**

### CODE REVIEW

File: heka/sandbox/filters/firefox_channel_switching.lua [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+Firefox Channel Switching
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [FireFoxChannelSwitching]
```
Comment (mreid-moz) [inline heka/sandbox/filters/firefox_channel_switching.lua]: Firefox

File: heka/sandbox/filters/firefox_channel_switching.lua [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    type = "SandboxFilter"
+    filename = "lua_filters/firefox_channel_switching.lua"
+    message_matcher = "Type == 'telemetry' && Fields[docType] == 'main' && Fields[appName] == 'Firefox'" # don't restrict by vendor
+    memory_limit = 1000000000
+    ticker_interval = 60
+    preserve_data = true
+        [FireFoxChannelSwitching.config]
```
Comment (mreid-moz) [inline heka/sandbox/filters/firefox_channel_switching.lua]: Firefox

File: heka/sandbox/filters/firefox_channel_switching.lua [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+-- l.C"esr" * -1 +
+-- l.P"esr-cck-" / "esr-partner" +
+l.Cc"other"
+
+function process_message()
+    local cid = read_message("Fields[clientId]")
+    if not cid then return -1, "missing clientId" end
+
+    local chan = read_message("Fields[appUpdateChannel]")
+    if not cid then return -1, "missing appUpdateChannel" end
```
Comment (mreid-moz) [inline heka/sandbox/filters/firefox_channel_switching.lua]: `if not chan`

Comment (trink) [inline heka/sandbox/filters/firefox_channel_switching.lua]: Do we want that check or should we default it to 'other'

Comment (mreid-moz) [inline heka/sandbox/filters/firefox_channel_switching.lua]: The check is good, we just need to check `chan` instead of `cid`.

**Pull request merged**